#### Lab Assignment 1 - Jack Travers
#### Student ID: 200013389
#### GitHub URL: https://github.com/Jacktravers81/PY4SA23_Assignment

##### Section One: Python Basics

Task 1: Create an If...Else statement that will test whether a number is divisible by three.

In [79]:
def divisible (n):
    if n % 3 == 0:
        print (n, "is divisible by 3")
    else:
        print (n, "is not divisible by 3")
divisible (11)

11 is not divisible by 3


Task 2: Create an If...Else statement that will test whether a type of fruit, represented as a text string, is in a list of acceptable fruits (apple, orange, pear, kiwi, or strawberry).

In [43]:
fruit = ["apple", "orange", "pear", "kiwi", "strawberry"]

def library(check):
    if check in fruit:
        print (check, "is acceptable")
    else:
        print (check, "is not acceptable")

library("berry")

berry is not acceptable


Task 3: Create a function to calculate the distance between two coordinates using the haversine
formula. Write the following formula where the input parameters are a pair of coordinates as two
lists.

In [95]:
# COME BACK TO ME!!!!! 
import math

a = sin²(φB - φA/2) + cos φA * cos φB * sin²(λB - λA/2)
c = 2 * atan2( √a, √(1−a) )
d = R ⋅ c

SyntaxError: invalid character '²' (U+00B2) (2577503932.py, line 4)

##### Section Two: Pandas and NumPy

Question 1: How many trees are of the Quercus or Acer genus?

In [79]:
import numpy as np 
import pandas as pd 
trees_df = pd.read_csv("Assignment 2 Data\portland_park_trees.csv", index_col=30)

trees_df.value_counts("Genus")

Genus
Pseudotsuga      6783
Acer             3586
Quercus          2089
Prunus           1333
Pinus             983
                 ... 
Eucommia            1
Eriobotrya          1
Daphniphyllum       1
Schima              1
Illicium            1
Length: 117, dtype: int64

Answer:
*Acer*: 3586,
*Quercus*: 2089

Question 2: : How many trees are of the Quercus or Acer genus and have a DBH larger than 50 inches?

In [91]:
import numpy as np 
import pandas as pd 
trees_df = pd.read_csv("Assignment 2 Data\portland_park_trees.csv")

just_acer = trees_df[(trees_df["Genus"]=="Acer") & (trees_df["DBH"]>50)][["Genus", "DBH"]]
print(len(just_acer))

just_quercus = trees_df[(trees_df["Genus"]=="Quercus") & (trees_df["DBH"]>50)][["Genus", "DBH"]]
print(len(just_quercus))


44
80


Answer: *Acer*: 44, *Quercus*: 80

Question 3: Which genus has the highest mean DBH of the following genera: Quercus, Acer, or Fraxinus? 

In [89]:
import numpy as np 
import pandas as pd 
trees_df = pd.read_csv("Assignment 2 Data\portland_park_trees.csv", sep=",", header=0, encoding="ISO-8859-1")
trees_df.columns = [column.replace(" ", "_") for column in trees_df.columns]
tree_stats = pd.DataFrame()
tree_stats ["Mean"] = trees_df.groupby("Genus")['DBH'].mean()
tree_stats

acer_stats = trees_df[trees_df["Genus"]=="Acer"][["Genus", "DBH"]].mean()
quercus_stats = trees_df[trees_df["Genus"]=="Quercus"][["Genus", "DBH"]].mean()
fraxinus_stats = trees_df[trees_df["Genus"]=="Fraxinus"][["Genus", "DBH"]].mean()

print(acer_stats)
print(quercus_stats)
print(fraxinus_stats)


DBH    18.419085
dtype: float64
DBH    23.568238
dtype: float64
DBH    11.03361
dtype: float64


C:\Users\User\AppData\Local\Temp\ipykernel_18164\1292205718.py:9: FutureWarning: The default value of numeric_only in DataFrame.mean is deprecated. In a future version, it will default to False. In addition, specifying 'numeric_only=None' is deprecated. Select only valid columns or specify the value of numeric_only to silence this warning.
  acer_stats = trees_df[trees_df["Genus"]=="Acer"][["Genus", "DBH"]].mean()
C:\Users\User\AppData\Local\Temp\ipykernel_18164\1292205718.py:10: FutureWarning: The default value of numeric_only in DataFrame.mean is deprecated. In a future version, it will default to False. In addition, specifying 'numeric_only=None' is deprecated. Select only valid columns or specify the value of numeric_only to silence this warning.
  quercus_stats = trees_df[trees_df["Genus"]=="Quercus"][["Genus", "DBH"]].mean()
C:\Users\User\AppData\Local\Temp\ipykernel_18164\1292205718.py:11: FutureWarning: The default value of numeric_only in DataFrame.mean is deprecated. In a fut

Answer: *Acer* mean DBH: 18.4, *Quercus* mean DBH: 23.6, *Fraxinus* mean DBH: 11

Question 4: How many different species of trees are recorded in the Acer genus?

In [93]:
just_acer["Species"].nunique()

20

Answer: Different *Acer* species: 20

Question 5: cities.csv dataframe tasks

• Calculate a new column named "pop_M" (population in millions), by transforming the
"pop" (population) column

In [121]:
import pandas as pd 
cities_df = pd.read_csv("Assignment 2 Data\world_cities.csv")
cities_df["pop_M"] = (cities_df["pop"]/1000000)

city country    lat    lon  capital     pop_M
33285        San'a   Yemen  15.38  44.21        1  1.921589
36905       Ta'izz   Yemen  13.60  44.04        0  0.690689
43305  al-Hudaydah   Yemen  14.80  42.95        0  0.677315
390           Aden   Yemen  12.79  45.03        0  0.625361
43432   al-Mukalla   Yemen  14.54  49.13        0  0.307033

• Remove the original "pop" column

In [ ]:
cities_df.drop("pop", axis=1, inplace=True)
cities_j = cities_df[cities_df.city.str.startswith('J')]
cities_j.head(1)

• Choose/subset a city that starts with the same letter as your first name (for example,
"Mexico City" if your first name is Michael)

In [ ]:
cities_j = cities_df[cities_df.city.str.startswith('J')]
cities_j.head(1)

• Subset the five biggest (i.e., largest population sizes) cities from the country where your
selected city is

In [ ]:
(cities_df.query("country == 'Yemen'")
.sort_values("pop_M", ascending=False)
.head(5))

##### Section Three: Python Data Visualization

Graph 1: Create a scatterplot for just trees in the Ulmus genus with DBH mapped to the x-axis
and tree height mapped to the y-axis (Hint: You will need to use the “Genus”, “DBH”, and
“TreeHeight” attributes.). 